app/jobs/multi_region_kms_migration/user_migration_job.rb

```diff
@@ -0,0 +1,73 @@
+module MultiRegionKmsMigration
+  class UserMigrationJob < ApplicationJob
+    MAXIMUM_ERROR_TOLERANCE = 10
+
+    include ::NewRelic::Agent::MethodTracer
+
+    def perform(statement_timeout: 120, user_count: 1000)
+      return unless IdentityConfig.store.multi_region_kms_migration_jobs_enabled
+
+      error_count = 0
+      success_count = 0
+
+      users = find_users_to_migrate(statement_timeout:, user_count:)
+      users.each do |user|
+        return if error_count >= MAXIMUM_ERROR_TOLERANCE # rubocop:disable Lint/NonLocalExitFromIterator
+
+        Encryption::MultiRegionKmsMigration::UserMigrator.new(user).migrate!
+        success_count += 1
+        analytics.multi_region_kms_migration_user_migrated(
+          success: true,
+          user_id: user.id,
+          exception: nil,
+        )
+      rescue => err
+        error_count += 1
+        analytics.multi_region_kms_migration_user_migrated(
+          success: false,
+          user_id: user.id,
+          exception: err.inspect,
+        )
+      end
+      analytics.multi_region_kms_migration_user_migration_summary(
+        user_count: users.size,
+        success_count: success_count,
+        error_count: error_count,
+      )
+    end
+
+    def find_users_to_migrate(statement_timeout:, user_count:)
+      User.transaction do
```

Comment:
Is this in a transaction to keep the statement_timeout confined to the transaction scope?

Reply:
That is exactly correct. That allows us to use `SET LOCAL` below. From the [postgresql docs](https://www.postgresql.org/docs/current/sql-set.html):

> The effects of SET LOCAL last only till the end of the current transaction, whether committed or not.